Intellij 2016.3 Caused by: groovy.lang.GroovyRuntimeException: replaceNode() can only be used to replace a single node.

Hi,

When importing a gradle project into Intellij 2013.3 i get the error
"Caused by: groovy.lang.GroovyRuntimeException: replaceNode() can only be used to replace a single node."

Looks at the code it looks like line 202 is at fault:
`compilerConfiguration.annotationProcessing.replaceNode{`
I'm guessing IntelliJ 2016.3 has changed so a newly created project does not contain `<annotationProcessing>`
`</annotationProcessing>`
which means compilerConfiguration.annotationProcessing has a count of 0?
If i manually add that to the .idea\compiler.xml file then the loading of the gradle project works.

--- a/src/main/groovy/org/inferred/gradle/ProcessorsPlugin.groovy
+++ b/src/main/groovy/org/inferred/gradle/ProcessorsPlugin.groovy
@@ -199,6 +199,10 @@ class ProcessorsPlugin implements Plugin<Project> {
       throw new GradleException("Unable to find CompilerConfiguration element")
     }
 
+    if (compilerConfiguration.annotationProcessing.isEmpty()) {
+      new Node(compilerConfiguration, "annotationProcessing")
+    }
+
     compilerConfiguration.annotationProcessing.replaceNode{
       annotationProcessing() {
         profile(default: 'true', name: 'Default', enabled: 'true') {

--- a/src/test/groovy/org/inferred/gradle/ProcessorsPluginFunctionalTest.groovy
+++ b/src/test/groovy/org/inferred/gradle/ProcessorsPluginFunctionalTest.groovy
@@ -573,6 +573,93 @@ public class ProcessorsPluginFunctionalTest {
   }
 
   @Test
+  public void testAnnotationProcessingInIdeaCompilerXml() throws IOException {
+    buildFile << """
+      apply plugin: 'java'
+      apply plugin: 'idea'
+      apply plugin: 'org.inferred.processors'
+    """
+
+    new File(testProjectDir.newFolder('.idea'), 'compiler.xml') << """
+      <?xml version="1.0" encoding="UTF-8"?>
+      <project version="4">
+        <component name="CompilerConfiguration">
+          <annotationProcessing/>
+        </component>
+      </project>
+    """.trim()
+
+    File testProjectDirRoot = testProjectDir.getRoot()
+    GradleRunner.create()
+            .withProjectDir(testProjectDirRoot)
+            .withArguments("idea", "--stacktrace")
+            .build()
+
+    def xml = new XmlSlurper().parse(testProjectDirRoot.toPath().resolve(".idea/compiler.xml").toFile())
+
+    def expected = new XmlSlurper().parseText("""
+      <?xml version="1.0" encoding="UTF-8"?>
+      <project version="4">
+        <component name="CompilerConfiguration">
+          <annotationProcessing>
+            <profile default="true" name="Default" enabled="true">
+              <sourceOutputDir name="generated_src"/>
+              <sourceTestOutputDir name="generated_testSrc"/>
+              <outputRelativeToContentRoot value="true"/>
+              <processorPath useClasspath="true"/>
+            </profile>
+          </annotationProcessing>
+        </component>
+      </project>
+    """.trim())
+
+    assertEquals(expected, xml)
+  }
+
+  @Test
+  public void testNoAnnotationProcessingInIdeaCompilerXml() throws IOException {
+    buildFile << """
+      apply plugin: 'java'
+      apply plugin: 'idea'
+      apply plugin: 'org.inferred.processors'
+    """
+
+    new File(testProjectDir.newFolder('.idea'), 'compiler.xml') << """
+      <?xml version="1.0" encoding="UTF-8"?>
+      <project version="4">
+        <component name="CompilerConfiguration">
+        </component>
+      </project>
+    """.trim()
+
+    File testProjectDirRoot = testProjectDir.getRoot()
+    GradleRunner.create()
+            .withProjectDir(testProjectDirRoot)
+            .withArguments("idea", "--stacktrace")
+            .build()
+
+    def xml = new XmlSlurper().parse(testProjectDirRoot.toPath().resolve(".idea/compiler.xml").toFile())
+
+    def expected = new XmlSlurper().parseText("""
+      <?xml version="1.0" encoding="UTF-8"?>
+      <project version="4">
+        <component name="CompilerConfiguration">
+          <annotationProcessing>
+            <profile default="true" name="Default" enabled="true">
+              <sourceOutputDir name="generated_src"/>
+              <sourceTestOutputDir name="generated_testSrc"/>
+              <outputRelativeToContentRoot value="true"/>
+              <processorPath useClasspath="true"/>
+            </profile>
+          </annotationProcessing>
+        </component>
+      </project>
+    """.trim())
+
+    assertEquals(expected, xml)
+  }
+
+  @Test
   public void testOnlyApplyToSubProject() {
     testProjectDir.newFolder("projectA")
     testProjectDir.newFolder("projectB")

